projects/hathor-wallet-headless/0003-fireblocks-integration.md

```diff
@@ -0,0 +1,130 @@
+# Summary
+[summary]: #summary
+
+Hathor headless wallet integration with Fireblocks using RAW Signing API.
+
+# Motivation
+[motivation]: #motivation
+
+Fireblocks manages the private key so removing the private key from the headless instance will allow users to run the headless in a more secure manner.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+<details>
+
+ <summary><code>POST</code> <code><b>/wallet/fireblocks/start</b></code> <code>(Start a wallet with the fireblocks integration)</code></summary>
+
+This API will use the wallet's `setExternalTxSigningMethod` to register a method that will start a client for fireblocks and use Fireblock's RAW signing API to sign the transaction.
+
+##### Parameters
+
+> | name | type | data type | description | location |
+> | --- | --- | --- | --- | --- |
+> | xpub-id | required | string | The id of the xpub in the config | body | 
+> | wallet-id | required | string | create a wallet with this id | body |
+> | raw | required | boolean | use RAW signing | body |

```

Comment:
The fireblocks integration has other ways of working, we are only adding support for raw signing so this parameter is just so we are clear on how the headless will work the integration.

Since it is actually useless for now I think we can remove it, what so you think?

Reply:
I agree on removing it.

We could leave only a note mentioning this other possibility, but remove it from the main proposal of the design.

Reply:
Done